infrastructure/push-server/src/main.ts

```diff
@@ -0,0 +1,305 @@
+import { Construct } from 'constructs';
+import {
+  App,
+  DataTerraformRemoteState,
+  S3Backend,
+  TerraformStack,
+} from 'cdktf';
+import { config } from './config';
+import {
+  PocketECSApplication,
+  PocketPagerDuty,
+} from '@pocket-tools/terraform-modules';
+import * as fs from 'fs';
+
+import { AwsProvider } from '@cdktf/provider-aws/lib/provider';
+import { DataAwsRegion } from '@cdktf/provider-aws/lib/data-aws-region';
+import { DataAwsCallerIdentity } from '@cdktf/provider-aws/lib/data-aws-caller-identity';
+import { CloudwatchLogGroup } from '@cdktf/provider-aws/lib/cloudwatch-log-group';
+import { DataAwsKmsAlias } from '@cdktf/provider-aws/lib/data-aws-kms-alias';
+import { DataAwsSqsQueue } from '@cdktf/provider-aws/lib/data-aws-sqs-queue';
+import { DataAwsSnsTopic } from '@cdktf/provider-aws/lib/data-aws-sns-topic';
+import { PagerdutyProvider } from '@cdktf/provider-pagerduty/lib/provider';
+import { NullProvider } from '@cdktf/provider-null/lib/provider';
+import { LocalProvider } from '@cdktf/provider-local/lib/provider';
+import { ArchiveProvider } from '@cdktf/provider-archive/lib/provider';
+
+class PushServer extends TerraformStack {
+  config: any;
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    new AwsProvider(this, 'aws', { region: 'us-east-1' });
+    new PagerdutyProvider(this, 'pagerduty_provider', { token: undefined });
+    new NullProvider(this, 'null-provider');
+    new LocalProvider(this, 'local-provider');
+    new ArchiveProvider(this, 'archive-provider');
+
+    new S3Backend(this, {
+      bucket: `mozilla-pocket-team-${config.environment.toLowerCase()}-terraform-state`,
+      dynamodbTable: `mozilla-pocket-team-${config.environment.toLowerCase()}-terraform-state`,
+      key: config.name,
+      region: 'us-east-1',
+    });
+
+    const region = new DataAwsRegion(this, 'region');
+    const caller = new DataAwsCallerIdentity(this, 'caller');
+
+    //NOTE: THis service uses CPU based autoscaling, this should move to SQS based autoscaling based on the Job queue in the future.
```

Comment:
Looking at its past scaling events, converting this to be CPU bound scaling should be fine for a while. This service has backed up before and handled through the messages.

Reply:
Once we have this, we could convert some of the SQS lambdas to ecs tasks which are easier to test and manage too.